sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1229,3 +1229,98 @@ case class Flatten(child: Expression) extends UnaryExpression {
 
   override def prettyName: String = "flatten"
 }
+
+/**
+ * Returns the array containing the given input value (left) count (right) times.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(element, count) - Returns the array containing element count times.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('123', 2);
+       ['123', '123']
+  """)
+case class ArrayRepeat(left: Expression, right: Expression)
+  extends BinaryExpression {
+
+  override def dataType: ArrayType = ArrayType(left.dataType, left.nullable)
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val expected = IntegerType
+    if (!expected.acceptsType(right.dataType)) {
+      val mismatch = s"argument 2 requires ${expected.simpleString} type, " +
+        s"however, '${right.sql}' is of ${right.dataType.simpleString} type."
+      TypeCheckResult.TypeCheckFailure(mismatch)
+    } else {
+      TypeCheckResult.TypeCheckSuccess
+    }
+  }
+
+  override def nullable: Boolean = false
+
+  override def eval(input: InternalRow): Any = {
+    new GenericArrayData(List.fill(right.eval(input).asInstanceOf[Integer])(left.eval(input)))
+  }
+
+  override def prettyName: String = "array_repeat"
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+
+    val leftGen = left.genCode(ctx)
+    val rightGen = right.genCode(ctx)
+    val element = leftGen.value
+    val count = rightGen.value
+    val et = dataType.elementType
+    val isPrimitive = CodeGenerator.isPrimitiveType(et)
+
+    val arrayDataName = ctx.freshName("arrayData")
+    val arrayName = ctx.freshName("arrayObject")
+    val initialization = (numElements: String) => if (isPrimitive) {
+      val arrayName = ctx.freshName("array")
+      val baseOffset = Platform.BYTE_ARRAY_OFFSET
+      s"""
+         | int numBytes = ${et.defaultSize} * $numElements;
+         | int unsafeArraySizeInBytes = UnsafeArrayData.calculateHeaderPortionInBytes($numElements)
+         |   + org.apache.spark.unsafe.array.ByteArrayMethods
+         |     .roundNumberOfBytesToNearestWord(numBytes);
+         | byte[] $arrayName = new byte[unsafeArraySizeInBytes];
```

Comment:
Do we need size check? For example, 0x7000_0000 long elements do not fit into `byte[]`.

Reply:
Would this not just throw an exception anyway if unsafeArraySizeInBytes is greater than 2^31-1?